src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +386,55 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
 	}
+
+	/**
+	 * Confirm the Inventory rows of inventory.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
+	 * @throws OHDataValidationException
+	 */
+	public void confirmMedicalInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		// validate the inventory
+		this.validateMedicalInventoryRow(inventory, inventoryRowSearchList);
+		// create Movements
+		String dischargeCode = inventory.getDischargeType();
+		String chargeCode = inventory.getChargeType();
+		Integer supplierId = inventory.getSupplier();
+		String wardCode = inventory.getDestination();
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(chargeCode);
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(dischargeCode);
+		Supplier supplier = supplierManager.getByID(supplierId);
+		Ward ward = wardManager.findWard(wardCode);
+		LocalDateTime today = LocalDateTime.now();
+		for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+			String reference = "Ref-" + LocalDateTime.now();
+			MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+			double theoQty = medicalInventoryRow.getTheoreticQty();
+			double realQty = medicalInventoryRow.getRealQty();
+			Double ajustQty = theoQty - realQty;
+			Medical medical = medicalInventoryRow.getMedical();
+			String lotCode = medicalInventoryRow.getLot().getCode();
+			Lot currentLot = movStockInsertingManager.getLot(lotCode);
+			boolean isNew = medicalInventoryRow.isNewLot();
+			if (realQty > theoQty && !isNew) { // charge movement when realQty > theoQty
+				Movement movement = new Movement(medical, chargeType, null, currentLot, today, -(ajustQty.intValue()), supplier, reference);
+				List<Movement> chargeMovement = new ArrayList<>();
+				chargeMovement.add(movement);
+				chargeMovement = movStockInsertingManager.newMultipleChargingMovements(chargeMovement, reference);
+			} else if (realQty < theoQty && !isNew) { // discharge movement when realQty < theoQty
+				Movement movement = new Movement(medical, dischargeType, ward, currentLot, today, ajustQty.intValue(), null, reference);
+				List<Movement> dischargeMovement = new ArrayList<>();
+				dischargeMovement.add(movement);
+				dischargeMovement = movStockInsertingManager.newMultipleDischargingMovements(dischargeMovement, reference);
+			}
+			// If new lot has been inserted, create charge movement with real qty
+			if (isNew) {
+				Movement movement = new Movement(medical, chargeType, null, currentLot, today, (int) realQty, supplier, reference);
+				List<Movement> chargeMovement = new ArrayList<>();
+				chargeMovement.add(movement);
+				chargeMovement = movStockInsertingManager.newMultipleChargingMovements(chargeMovement, reference);
+			}
```

Comment:
This should not be necessary because new lots have been already created at this point so the only thing to do is to use them, am I wrong?
```suggestion
			if (realQty > theoQty) { // charge movement when realQty > theoQty
				Movement movement = new Movement(medical, chargeType, null, currentLot, today, -(ajustQty.intValue()), supplier, reference);
				List<Movement> chargeMovement = new ArrayList<>();
				chargeMovement.add(movement);
				chargeMovement = movStockInsertingManager.newMultipleChargingMovements(chargeMovement, reference);
			} else { // discharge movement when realQty < theoQty
				Movement movement = new Movement(medical, dischargeType, ward, currentLot, today, ajustQty.intValue(), null, reference);
				List<Movement> dischargeMovement = new ArrayList<>();
				dischargeMovement.add(movement);
				dischargeMovement = movStockInsertingManager.newMultipleDischargingMovements(dischargeMovement, reference);
			}
```

Reply:
yes the lot is create but it is not link to the movement.

Reply:
the lot is linked to a movement within the movement itself, not in any other way...

Reply:
please when you doing charge movement on main store with new lot you create first the new lot and then link the lot to the movement right ? but why here we just create lot without  link that lot to movement ?

Reply:
the charge movement will be done in the first `if` if you remove the condition `isNew`

Reply:
okay